### PR TITLE
VMAccess support for SSH2 public keys

### PR DESCRIPTION
**Pull request by mayankdaruka-msft**
- Logic to convert SSH2 keys to OpenSSH keys (ssh-rsa)
- Allows PuTTYgen generated keys to be used as public keys

### CODE REVIEW

File: VMAccess/vmaccess.py [hunk 1]
```diff
@@ -275,6 +276,15 @@ def _set_user_account_pub_key(protect_settings, hutil):
 
     # Reset ssh key with the new public key passed in or reuse old public key.
     if cert_txt:
+        # support for SSH2-compatible format for public keys in addition to OpenSSH-compatible format
```
Comment (deeptivaidymsft) [inline VMAccess/vmaccess.py:279]: You might want to add to the unit test in test_reset_account.py to cover your changes.

File: VMAccess/vmaccess.py [hunk 2]
```diff
@@ -275,6 +276,15 @@ def _set_user_account_pub_key(protect_settings, hutil):
...
+        if (cert_txt.strip().startswith(BeginSSHTag)):
+            ext_utils.set_file_contents("temp.pub", cert_txt.strip())
+            retcode, output = ext_utils.run_command_get_output(['ssh-keygen', '-i', '-f', 'temp.pub'])
+            if retcode > 0:
+                raise Exception("Failed to convert SSH2 key to OpenSSH key.")
+            cert_txt = output
+            os.remove("temp.pub")
+
```
Comment (deeptivaidymsft) [inline VMAccess/vmaccess.py:287]: nit: might be helpful to add logging message here that we've successfully converted the SSH2 key to OpenSSH

File: VMAccess/vmaccess.py [hunk 3]
```diff
@@ -275,6 +276,16 @@ def _set_user_account_pub_key(protect_settings, hutil):
 
     # Reset ssh key with the new public key passed in or reuse old public key.
     if cert_txt:
+        # support for SSH2-compatible format for public keys in addition to OpenSSH-compatible format
+        if (cert_txt.strip().startswith(BeginSSHTag)):
+            ext_utils.set_file_contents("temp.pub", cert_txt.strip())
```
Comment (norakoiralamsft) [inline VMAccess/vmaccess.py:281]: Will writing the key to a file be a security concern? Is there any other way to pass in the key to ssh-keygen command?

Comment (norakoiralamsft) [reply VMAccess/vmaccess.py:281]: Actually, writing the key into the command line is also a concern, I think writing to file might be best way to go. Let me know if you have any other ideas.

**Review by norakoiralamsft — APPROVED** [review]

**Review by norakoiralamsft — APPROVED** [review]

**Review by deeptivaidymsft — APPROVED** [review]

File: VMAccess/vmaccess.py [hunk 4]
```diff
@@ -275,6 +276,16 @@ def _set_user_account_pub_key(protect_settings, hutil):
...
+            retcode, output = ext_utils.run_command_get_output(['ssh-keygen', '-i', '-f', 'temp.pub'])
```
Comment (jscalev) [inline VMAccess/vmaccess.py:282]: Is ssh-keygen available on all distros, or do we need to check for existence of it first?

**Pull request merged**